cli/azd/pkg/project/scaffold_gen.go

```diff
@@ -0,0 +1,241 @@
+}
+
+// Generates the filesystem of all infrastructure files to be placed, rooted at the project directory.
+// The content only includes `./infra` currently.
+func infraFsForProject(ctx context.Context, prjConfig *ProjectConfig) (fs.FS, error) {
+	infraFS, err := infraFs(ctx, prjConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	infraPathPrefix := DefaultPath
+	if prjConfig.Infra.Path != "" {
+		infraPathPrefix = prjConfig.Infra.Path
+	}
+
+	// root the generated content at the project directory
+	generatedFS := memfs.New()
+	err = fs.WalkDir(infraFS, ".", func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if d.IsDir() {
+			return nil
+		}
+
+		err = generatedFS.MkdirAll(filepath.Join(infraPathPrefix, filepath.Dir(path)), osutil.PermissionDirectoryOwnerOnly)
+		if err != nil {
+			return err
+		}
+
+		contents, err := fs.ReadFile(infraFS, path)
+		if err != nil {
+			return err
+		}
+
+		return generatedFS.WriteFile(filepath.Join(infraPathPrefix, path), contents, d.Type().Perm())
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return generatedFS, nil
+}
+
+func infraSpec(projectConfig *ProjectConfig) (*scaffold.InfraSpec, error) {
+	infraSpec := scaffold.InfraSpec{}
+	// backends -> frontends
+	backendMapping := map[string]string{}
+
+	for _, res := range projectConfig.Resources {
+		switch res.Type {
+		case ResourceTypeDbRedis:
+			infraSpec.DbRedis = &scaffold.DatabaseRedis{}
+		case ResourceTypeDbMongo:
+			infraSpec.DbCosmosMongo = &scaffold.DatabaseCosmosMongo{
+				DatabaseName: res.Name,
+			}
+		case ResourceTypeDbPostgres:
+			infraSpec.DbPostgres = &scaffold.DatabasePostgres{
+				DatabaseName: res.Name,
+				DatabaseUser: "pgadmin",
+			}
+		case ResourceTypeHostContainerApp:
+			svcSpec := scaffold.ServiceSpec{
+				Name: res.Name,
+				Port: -1,
+			}
+
+			err := mapContainerApp(res, &svcSpec)
+			if err != nil {
+				return nil, err
+			}
+
+			err = mapHostUses(res, &svcSpec, backendMapping, projectConfig)
+			if err != nil {
+				return nil, err
+			}
+
+			infraSpec.Services = append(infraSpec.Services, svcSpec)
+		}
+	}
+
+	// create reverse frontends -> backends mapping
+	for _, svc := range infraSpec.Services {
+		if front, ok := backendMapping[svc.Name]; ok {
+			if svc.Backend == nil {
+				svc.Backend = &scaffold.Backend{}
+			}
+
+			svc.Backend.Frontends = append(svc.Backend.Frontends, scaffold.ServiceReference{Name: front})
+		}
+	}
+
+	slices.SortFunc(infraSpec.Services, func(a, b scaffold.ServiceSpec) int {
+		return strings.Compare(a.Name, b.Name)
+	})
+
+	return &infraSpec, nil
+}
+
+func mapContainerApp(res *ResourceConfig, svcSpec *scaffold.ServiceSpec) error {
+	props := res.Props.(ContainerAppProps)
+	for _, envVar := range props.Env {
+		isSecret := len(envVar.Secret) > 0
+		value := envVar.Value
+		if isSecret {
+			// TODO: handle secrets
+			continue
+		}
+
+		svcSpec.Env[envVar.Name] = value
+	}
+
+	port := props.Port
+	if port < 1 || port > 65535 {
```

Comment:
nit: consider moving 65535 out to a constant.

Reply:
I will leave it as-is for now -- we could think about creating some sort of a networking package that validates port numbers amongst other things. But creating a constant right now would end up adding to global variables in `project` that wouldn't quite be consumable either, so I'm pretty happy with how it looks right now.